ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/editorconfig/TextEditorConfiguration.java

```diff
@@ -97,4 +98,11 @@
      */
     @Nullable
     ChangeInterceptorProvider getChangeInterceptorProvider();
+
+    /**
+     * Return the {@link SignatureHelpProvider}
+     * @return the signature help provider
+     */
+    @Nullable
+    SignatureHelpProvider getSignatureHelpProvider();
```

Comment:
other new interface like SignatureHelp is using `Optional<>`
shouldn't we use `Optional<SignatureHelpProvider>` here as well ?